src/ModuleList/Lesson/useLesson.ts

```diff
@@ -0,0 +1,41 @@
+import { useEffect } from 'react';
+import { useImmerReducer } from 'use-immer';
+
+import { Action } from '../../common/types';
+import { Meta } from '../../common/api/curriculum/lesson/requests';
+import { useLessonMeta } from '../../common/api/curriculum/lesson';
+
+export interface LessonState {
+  numSlides: number;
+}
+
+export enum LessonActions {
+  SET_LESSON
+}
+
+type ReducerAction = Action<LessonActions, Meta | undefined>;
+
+function reducer(state: LessonState, action: ReducerAction) {
+  switch (action.type) {
+    case LessonActions.SET_LESSON:
+      state.numSlides = action.payload ? action.payload.numSlides : 0;
+      return;
+  }
+};
+
+
+const initialState: LessonState = {
+  numSlides: 0,
+};
+
+export function useLesson(lessonId: string) {
+const { data, isLoading } = useLessonMeta(Number(lessonId));
```

Comment:
Why do we pass in a string for `useLesson` and then cast it to a number on line 32? I think it would be better to accept a number for `useLesson` and then have whoever is using this hook cast their value to a number.

Reply:
This one self resolves because we get rid of this function